### PR TITLE
Task/merge sandbox and tenant controllers

### PR DESCRIPTION
**Pull request by esotrope**
not done yet but i wanted to get buy-in on these endpoints before i adjust the UI to use them.

### CODE REVIEW

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/service/TenantService.java [hunk 1]
```diff
@@ -20,24 +20,27 @@
      *
      * @return The Default information and a list of all existing tenants for this Instance
      */
+    @PreAuthorize("hasAuthority('PERM_TENANT_READ')")
```
Comment (esotrope) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/service/TenantService.java]: added security to the service and todoned that todo

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/service/TenantService.java [hunk 2]
```diff
@@ -46,20 +49,19 @@
...
     /**
      * @return a TenantConfiguration containing the default properties of the current tenant
      */
+    @PreAuthorize("hasAuthority('PERM_TENANT_READ')")
     TenantConfiguration getTenantDefaults();
 
-    /**
-     * @return a TenantConfiguration containing the default properties of the current tenant
-     */
-    TenantConfiguration getParentTenantDefaults(String key);
```
Comment (esotrope) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/service/TenantService.java]: there was no use of this in the UI

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/service/impl/DefaultTenantService.java [hunk 3]
```diff
@@ -68,21 +68,28 @@
     }
 
     @Override
-    public void createTenant(TenantConfiguration tenantConfiguration) {
+    public TenantConfiguration createTenant(final TenantConfiguration tenantConfiguration) {
```
Comment (esotrope) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/service/impl/DefaultTenantService.java]: this is out of convention - looks like right now the UI ignores what update/create return though - @ftdholbrook any reason we shouldn't return the result of update/create?

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/service/impl/DefaultTenantService.java [hunk 4]
```diff
@@ -68,21 +68,28 @@
...
+        // update the translationLocalization with specific
+        tenantConfiguration.getReporting().setTranslationLocation(
+                getTenantLocalization(
+                        tenantConfiguration.getLocalization(),
+                        tenantConfiguration.getReporting(),
+                        tenantConfiguration.getTenant().getKey()
+                )
+        );
+        tenantAdministrationRequestHandler.createTenant(tenantConfiguration);
+        return tenantConfiguration;
```
Comment (ftdholbrook) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/service/impl/DefaultTenantService.java]: a language that evaluates if statements as expressions makes this style if block an anti-pattern.  I know it is common in Java, but I don't care for it.

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/service/impl/DefaultTenantService.java (hunk 3 above)
Comment (ftdholbrook) [reply admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/service/impl/DefaultTenantService.java]: There is no result it is passed down stream to a message. This should be void

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/service/impl/DefaultTenantService.java [hunk 5]
```diff
@@ -94,7 +101,7 @@ public void deleteTenant(final String tenantKey) {
     }
 
     @Override
-    public TenantConfiguration updateTenant(TenantConfiguration tenantConfiguration) {
+    public TenantConfiguration updateTenant(final TenantConfiguration tenantConfiguration) {
```
Comment (ftdholbrook) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/service/impl/DefaultTenantService.java]: this should return void

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/web/TenantController.java [hunk 6]
```diff
@@ -3,58 +3,60 @@
...
+    TenantConfiguration get(@PathVariable("key") final String key) {
+        return service.findActiveTenantConfigurationByKey(key)
+                .orElseThrow(NoSuchElementException::new);
     }
 
     @PostMapping
     @ResponseStatus(value = HttpStatus.CREATED)
-    void create(@RequestBody final TenantConfiguration tenantConfiguration) {
-        tenantService.createTenant(tenantConfiguration);
+    TenantConfiguration post(@RequestBody final TenantConfiguration configuration) {
```
Comment (ftdholbrook) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/web/TenantController.java]: this should be void

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/web/TenantController.java [hunk 7]
```diff
@@ -3,58 +3,60 @@
...
+        return service.createTenant(configuration);
     }
 
     @PutMapping
     @ResponseStatus(value = HttpStatus.OK)
-    void put(@RequestBody final TenantConfiguration tenantConfiguration) {
-        tenantService.updateTenant(tenantConfiguration);
+    TenantConfiguration put(@RequestBody final TenantConfiguration configuration) {
```
Comment (ftdholbrook) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/web/TenantController.java]: dido: void

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/web/TenantController.java [hunk 8]
```diff
@@ -3,58 +3,60 @@
...
-    public TenantController(TenantService tenantService) {
-        this.tenantService = tenantService;
+    TenantController(final TenantService service) {
+        this.service = service;
     }
 
     @GetMapping
-    public Set<TenantConfiguration> get() {
-        return tenantService.getAll(TenantType.TENANT);
+    Set<TenantConfiguration> get(@RequestParam final TenantType type) {
```
Comment (esotrope) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/web/TenantController.java]: allows the client to specify if they want just sandboxes or just tenants or all - which isn't a case

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/web/TenantController.java [hunk 9]
```diff
@@ -3,58 +3,60 @@
...
     @GetMapping("/{key}")
-    TenantConfiguration getByKey(@PathVariable("key") final String key) {
-        final Optional<TenantConfiguration> activeTenantConfigurationByKey = tenantService.findActiveTenantConfigurationByKey(key);
-        if (activeTenantConfigurationByKey.isPresent()) {
-            return activeTenantConfigurationByKey.get();
-        }
-        throw new NoSuchElementException();
+    TenantConfiguration get(@PathVariable("key") final String key) {
+        return service.findActiveTenantConfigurationByKey(key)
+                .orElseThrow(NoSuchElementException::new);
```
Comment (ftdholbrook) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/web/TenantController.java]: nice much cleaner forgot about ::new

**Pull request merged**